### PR TITLE
modsecurity 3.0.10 (new formula)

### PR DESCRIPTION
**Pull request by flier**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/m/modsecurity.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  depends_on "yajl"
+
+  uses_from_macos "curl", since: :monterey
+  uses_from_macos "libxml2"
+
+  def install
+    libxml2 = "#{MacOS.sdk_path_if_needed}/usr"
+    libxml2 = Formula["libxml2"].opt_prefix if OS.linux?
+
+    system "./build.sh"
```
Comment (SMillerDev) [inline Formula/m/modsecurity.rb:25]: What does this one do?

File: Formula/m/modsecurity.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      "--with-pcre2=#{Formula["pcre2"].opt_prefix}",
+      "--with-yajl=#{Formula["yajl"].opt_prefix}",
+    ]
+
+    system "./configure", *args, *std_configure_args, "--disable-silent-rules"
+    system "make", "install"
+  end
+
+  test do
+    system "#{bin}/modsec-rules-check", "\"SecAuditEngine RelevantOnly\""
```
Comment (SMillerDev) [inline Formula/m/modsecurity.rb:42]: Can we check the output of this?

File: Formula/m/modsecurity.rb (hunk 1 above)
Comment (flier) [reply Formula/m/modsecurity.rb:25]:
``` build.sh
#!/bin/sh

rm -rf autom4te.cache
rm -f aclocal.m4
case `uname` in Darwin*) glibtoolize --force --copy ;;
  *) libtoolize --force --copy ;; esac
autoreconf --install
autoheader
automake --add-missing --foreign --copy --force-missing
autoconf --force
rm -rf autom4te.cache
```

Comment (SMillerDev) [reply Formula/m/modsecurity.rb:25]: Can we just replace it with `system "autoreconf", "--force", "--install", "--verbose"`?

File: Formula/m/modsecurity.rb (hunk 2 above)
Comment (flier) [reply Formula/m/modsecurity.rb:42]: I think the outputs and [return value](https://github.com/SpiderLabs/ModSecurity/blob/v3/master/tools/rules-check/rules-check.cc#L100) work the same way

```sh
$ modsec-rules-check "SecAuditEngine RelevantOnly"
 : SecAuditEngine RelevantOnly  --  Loaded 0 rules.
Test ok.
```

File: Formula/m/modsecurity.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      "--with-yajl=#{Formula["yajl"].opt_prefix}",
+    ]
+
+    system "./configure", *args, *std_configure_args, "--disable-silent-rules"
+    system "make", "install"
+  end
+
+  test do
+    output = shell_output("#{bin}/modsec-rules-check \"SecAuditEngine RelevantOnly\"")
+    assert_match(/Test ok/, output)
```
Comment (SMillerDev) [inline Formula/m/modsecurity.rb:43]: No need to use a regex here
```suggestion
    assert_match "Test ok", output
```

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@flier can you squash your commits into one with the commit msg as `modsecurity 3.0.10 (new formula)`? Thanks!

**Review by p-linnane — APPROVED** [review]

**Pull request merged**